internal/extract/extract.go

```diff
@@ -0,0 +1,127 @@
+// Package extract defines all the possible extraction methods that can be applied on a response.
+//
+// Signatures
+//
+// it is important for consistency that each extractor follow the same function signature:
+//  func(r format.RequestTest, resp *http.Response, env map[string]interface{})
+//
+//
+// this will allow easier refactoring or interfacing later on if this becomes necessary.
+package extract
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ladydascalie/litmus/format"
+	"github.com/ladydascalie/litmus/internal"
+	"github.com/ladydascalie/litmus/p"
+	"github.com/ladydascalie/litmus/pkg"
+	e "github.com/pkg/errors"
```

Comment:
pkg/errors has the same methods as the stdlib errors package, is I'd just remember the alias.